vernac/topfmt.ml

```diff
@@ -83,7 +83,11 @@ let get_depth_boxes () = Some (Format.pp_get_max_boxes !std_ft ())
 let set_depth_boxes v =
   Format.pp_set_max_boxes !std_ft (match v with None -> default | Some v -> v)
 
-let get_margin () = Some (Format.pp_get_margin !std_ft ())
+let get_margin0 () = Format.pp_get_margin !std_ft ()
+
+let () = Profile_tactic.set_get_printing_width get_margin0
```

Comment:
This seems wrong at two levels:
- it will ignore the printing width set at init time (for example when we query the terminal), also it won't respect the `Set Printing Width` command
- moreover this cannot work for IDEs, that for example want to set the width quite dynamically (due to windows resizes, etc...)

Reply:
This is the function used by the Printing Width option.

Reply:
You set it only once at module init time, this seems too early to me; moreover changes won't be picked up.

Reply:
We set a hook not the current value

Reply:
I see, that's pretty confusing, I'm not sure why a "hook" is needed here.

Reply:
How else would you do it?

Reply:
Never mind, I got very confused as I didn't realize Profile_ltac is now in `engine`, not in `Ltac_plugin`.